presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -120,9 +134,11 @@
     private volatile boolean metastoreKnownToSupportTableParamLikePredicate;
 
     @Inject
-    public ThriftHiveMetastore(MetastoreLocator metastoreLocator, ThriftHiveMetastoreConfig thriftConfig)
+    public ThriftHiveMetastore(MetastoreLocator metastoreLocator, ThriftHiveMetastoreConfig thriftConfig, HdfsEnvironment hdfsEnvironment)
     {
+        this.hdfsContext = new HdfsContext(new ConnectorIdentity(DEFAULT_METASTORE_USER, Optional.empty(), Optional.empty()));
```

Comment:
I'm not sure we should use a hard-coded `presto` user for the metastore. I realize we do this for Glue, but I have the same concerns there. Using the table owner seems better, as the files would be created using the table owner.

Reply:
How the table owner can be retrieved in the constructor ?

Reply:
The table owner is available from the `Table` object. However, we can now use the user from `HiveIdentity`.

Reply:
We don't have `Table` nor `HiveIdentity` in the constructor.